python/tskit/trees.py

```diff
@@ -2116,15 +2116,45 @@ def num_tracked_samples(self, u=None):
         u = self.virtual_root if u is None else u
         return self._ll_tree.get_num_tracked_samples(u)
 
-    # TODO document these traversal arrays
-    # https://github.com/tskit-dev/tskit/issues/1788
     def preorder(self, u=NULL):
+        """
+        Returns a numpy array of node ids. Starting at `u`, yield the current node,
```

Comment:
This isn't quite right I think. How about 

> Returns a number array of node ids in `preorder <https://en.wikipedia.org/wiki/Tree_traversal#Pre-order_(NLR)>`. If the node ``u`` the traversal is rooted at this node (and it will be the first element in the returned array). Otherwise, all nodes reachable from the tree roots will be returned.  See :ref:`tutorials:sec_analysing_trees_traversals` for examples.

I don't think need to describe what "preorder" is, the wikipedia link is fine. I think talking about "yielding" here is more confusing, since this isn't a generator.

Reply:
Fixed